### DISCUSSION TITLE
Pandas recommendations for Xarray

### DISCUSSION THREAD

**Discussion by mgorfer** [astral-sh/ruff#7183] [Q&A]
I am getting `Pandas` warnings for `Xarray` Objects, which are just wrong.

![image](https://github.com/astral-sh/ruff/assets/13124998/128b0f52-c9a9-4a38-a900-9d981de9fcc1)

Is this a setup mistake on my side, or some other issue?

**Comment by hamedrabah** [comment]
This is not a setup error. The `PD` rules come from pandas-vet and are largely syntax/name based; Ruff does not infer that the receiver is an Xarray object before flagging methods/properties that look like the pandas patterns.

If the project uses Xarray heavily, the practical fix is to disable the offending `PDxxx` rule globally or with a per-file ignore, for example:

```toml
[tool.ruff.lint.per-file-ignores]
"src/xarray_code/**/*.py" = ["PD011"]
```

For isolated cases, use `# noqa: PD011` on the line. Prefer the specific code rather than ignoring all `PD` rules, since some pandas-vet checks may still be useful elsewhere. Without type-aware analysis, Ruff cannot reliably distinguish `DataArray.values` from `DataFrame.values` at that call site.
